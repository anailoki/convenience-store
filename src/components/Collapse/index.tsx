import { Minus, Plus } from '@phosphor-icons/react';
import React, { useState } from 'react';

interface ICollapseProps {
  label: string;
  children: React.ReactNode;
  isOpen?: boolean;
}

const Collapse = ({ label, children, isOpen }: ICollapseProps) => {
  const [open, setOpen] = useState(isOpen);
  return (
    <>
      <div
        data-testid='collapse'
        className='bg-slate-100 py-3 px-2 flex gap-x-8 justify-between items-center cursor-pointer mb-3'
        onClick={() => setOpen(!open)}
      >
        <span className='uppercase text-md font-medium'>{label}</span>
        {open ? <Minus size={20} /> : <Plus size={20} />}
      </div>
      {open && (
        <div className='my-3 px-4' data-testid='children-collapse'>
          {children}
        </div>
      )}
    </>
  );
};

export default Collapse;
