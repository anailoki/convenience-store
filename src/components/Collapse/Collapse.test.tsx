import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import Collapse from '.';

describe('Render Collapse component', () => {
  it('should render close collapse', () => {
    const { getByText } = render(
      <Collapse isOpen={false} label='Categorias'>
        <h1>Todas Categorias</h1>
      </Collapse>
    );
    getByText('Categorias');
  });

  it('should render the children when collapse is open', async () => {
    const { container } = render(
      <Collapse label='Categorias' isOpen={true}>
        <h1>Todas Categorias...</h1>
      </Collapse>
    );
    expect(container).toMatchSnapshot();
  });
});
