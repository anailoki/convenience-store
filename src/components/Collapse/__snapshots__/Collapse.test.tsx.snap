// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render Collapse component > should render the children when collapse is open 1`] = `
<div>
  <div
    class="bg-slate-100 py-3 px-2 flex gap-x-8 justify-between items-center cursor-pointer mb-3"
    data-testid="collapse"
  >
    <span
      class="uppercase text-md font-medium"
    >
      Categorias
    </span>
    <svg
      fill="currentColor"
      height="20"
      viewBox="0 0 256 256"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224,128a8,8,0,0,1-8,8H40a8,8,0,0,1,0-16H216A8,8,0,0,1,224,128Z"
      />
    </svg>
  </div>
  <div
    class="my-3 px-4"
    data-testid="children-collapse"
  >
    <h1>
      Todas Categorias...
    </h1>
  </div>
</div>
`;
